MergeSort/list.c

```diff
@@ -0,0 +1,221 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include "list.h"
+#include <string.h>
+
+int maxNameLen = 50;
+int maxPhoneLen = 15;
+
+typedef struct Node {
+    char name[51];
+    char phone[16];
+    struct Node* next;
+} Node;
+
+typedef struct List {
+    Node* head;
+} List;
+
+List* createList(void) {
+    List* list = malloc(sizeof(List));
+    if (list == NULL) {
+        perror("Memory allocate error");
+        return NULL;
+    }
+    list->head = NULL;
+}
+
+int compare(char* first, char* second, const int length, int kind) {
+    for (int i = 0; i < length; ++i) {
+        int firstElement = first[i];
+        int secondElement = second[i];
+
+        if (firstElement == 0) {
+            if ((secondElement != 0) && kind == 2) {
+                return -1;
+            }
+            else {
+                return 0;
+            }
```

Comment:
Такие штуки лучше записывать тернарным оператором:
```suggestion
            return ((secondElement != 0) && kind == 2) ? -1 : 0;
```